tests/ensemble/test_blockwise.py

```diff
@@ -127,6 +128,10 @@ def test_soft_voting_array(self):
         score = clf.score(X, y)
         assert isinstance(score, float)
 
+    @pytest.mark.xfail(
+        DASK_EXPR_ENABLED,
```

Comment:
what's the traceback here?

Reply:
Oop, yes, here it is:

<details>
<summary>traceback</summary>

```
___________________________ TestBlockwiseVotingClassifier.test_soft_voting_frame ___________________________

self = <tests.ensemble.test_blockwise.TestBlockwiseVotingClassifier object at 0x7f4ddb961cc0>

    @pytest.mark.xfail(
        DASK_EXPR_ENABLED,
        reason="AssertionError da.utils.assert_eq(result, result2)",
    )
    def test_soft_voting_frame(self):
        X, y = dask_ml.datasets.make_classification(chunks=25)
        X = dd.from_dask_array(X)
        y = dd.from_dask_array(y)

        clf = dask_ml.ensemble.BlockwiseVotingClassifier(
            sklearn.linear_model.LogisticRegression(solver="lbfgs"),
            voting="soft",
            classes=[0, 1],
        )
        clf.fit(X, y)
        assert len(clf.estimators_) == 4

        X2, y2 = dask_ml.datasets.make_classification(chunks=20)
        X2 = dd.from_dask_array(X2)
        y2 = dd.from_dask_array(y2)

        result = clf.predict(X2)
        assert isinstance(result, da.Array)  # TODO(pandas-IO)
        assert result.dtype == np.dtype("int64")
        assert len(result.shape) == 1 and np.isnan(result.shape[0])
        assert result.numblocks == (y2.npartitions,)
        result_ = result.compute()
        assert result_.dtype == result.dtype
        assert result_.shape == (len(y2),)

        result = clf.predict_proba(X2)
        assert result.dtype == np.dtype("float64")
        assert len(result.shape) == 2
        assert np.isnan(result.shape[0])
        assert result.shape[1] == 2
        assert result.numblocks == (5, 1)

        score = clf.score(X, y)
        assert isinstance(score, float)

        # ndarray
        X3, y3 = dask.compute(X2, y2)
        result2 = clf.predict_proba(X3)
        assert isinstance(result2, np.ndarray)
>       da.utils.assert_eq(result, result2)

tests/ensemble/test_blockwise.py:175:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../mambaforge/envs/dask-ml/lib/python3.10/site-packages/dask/array/utils.py:316: in assert_eq
    a, adt, a_meta, a_computed = _get_dt_meta_computed(
../../mambaforge/envs/dask-ml/lib/python3.10/site-packages/dask/array/utils.py:271: in _get_dt_meta_computed
    _check_dsk(x.dask)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

dsk = HighLevelGraph with 7 layers.
<dask.highlevelgraph.HighLevelGraph object at 0x7f4dc83fe2f0>
 0. mappartitions-c686289e...12e7b41f5b630e
 5. mean_chunk-9f055cf26340b81f414dc25e514faaf9
 6. mean_agg-aggregate-51c123479254f9c7add98275db70fc9f


    def _check_dsk(dsk):
        """Check that graph is well named and non-overlapping"""
        if not isinstance(dsk, HighLevelGraph):
            return

        dsk.validate()
        assert all(isinstance(k, (tuple, str)) for k in dsk.layers)
        freqs = frequencies(concat(dsk.layers.values()))
        non_one = {k: v for k, v in freqs.items() if v != 1}
        key_collisions = set()
        # Allow redundant keys if the values are equivalent
        for k in non_one.keys():
            for layer in dsk.layers.values():
                try:
                    key_collisions.add(tokenize(layer[k]))
                except KeyError:
                    pass
>       assert len(key_collisions) < 2, non_one
E       AssertionError: {('from-dask-array-01e2356894d58ad2ed2f9be4def91514', 0): 4, ('from-dask-array-01e2356894d58ad2ed2f9be4def91514', 1): 4, ('from-dask-array-01e2356894d58ad2ed2f9be4def91514', 2): 4, ('from-dask-array-01e2356894d58ad2ed2f9be4def91514', 3): 4, ('from-dask-array-01e2356894d58ad2ed2f9be4def91514', 4): 4, ('normal-409d9d12a3e72f33efafb65118fc04e5', 0, 0): 4, ('normal-409d9d12a3e72f33efafb65118fc04e5', 1, 0): 4, ('normal-409d9d12a3e72f33efafb65118fc04e5', 2, 0): 4, ('normal-409d9d12a3e72f33efafb65118fc04e5', 3, 0): 4, ('normal-409d9d12a3e72f33efafb65118fc04e5', 4, 0): 4, ('from-dask-array-202c9b965a34e7b38c0b914972a41e6c', 0): 4, ('from-dask-array-202c9b965a34e7b38c0b914972a41e6c', 1): 4, ('from-dask-array-202c9b965a34e7b38c0b914972a41e6c', 2): 4, ('from-dask-array-202c9b965a34e7b38c0b914972a41e6c', 3): 4, ('from-dask-array-202c9b965a34e7b38c0b914972a41e6c', 4): 4}

../../mambaforge/envs/dask-ml/lib/python3.10/site-packages/dask/array/utils.py:222: AssertionError
```

</details>

Reply:
hm that seems fine